### DISCUSSION TITLE
Adjust face recognition after date of picture?

### DISCUSSION THREAD

**Discussion by johved** [photoprism/photoprism#2405] [Ideas]
Hello, I'm new here.
 I really love this project and I have just started to explore all the features. 
One thing that hit me when i loaded the entire family photo catalog into the library is that the algorithm has a hard time recognizing a "child version" and an "adult version" of a face to be the same person. I guess it's no wonder why this is the case, given the natural changes of age, but I was wondering if it would help the algorithm to adjust the different faces of a person acording to the age, or the date of the photograph in the meta data. In my case many of the photograph have correct dates going back to year 2000 with multiple persons growing up from a young age to be in their twenties now.  
I guess it would be hard to implement if a library cotains alot of older, scanned photos because the date would then often be incorrect.

**Comment by srett** [comment]
This could be a viable approach for e.g. DigiKam, where a dedicated model is trained for each face. It might make sense there to train 2 or 3 different models for different age ranges. PhotoPrism does it differently: All faces just get analyzed by a generic face detection algorithm which emits a "bunch of values" that describe that face. Then all faces that are similar enough according to these values become a cluster that you can name. So, if you have a set of photos of one person, and some are form when that person was 8 years old, and some are from when they were 30, chances are high you get two distinct clusters, since as you have noted, the face changes quite a bit over that time. But once you have assigned the same name (person) to both clusters, any further photos of that person from when they were around 8 or 30 (probably much broader for 30, for obvious reasons), the faces should get picked up just fine by those two clusters.
